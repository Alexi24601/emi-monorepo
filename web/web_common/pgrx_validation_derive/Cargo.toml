[package]
name = "pgrx_validation_derive"
version = "0.1.0"
edition = "2021"

[dependencies]
syn.workspace = true
quote.workspace = true
proc-macro2.workspace = true

[dev-dependencies]
trybuild.workspace = true
validation-errors.workspace = true

[lib]
proc-macro = true

[features]
pgrx = []
