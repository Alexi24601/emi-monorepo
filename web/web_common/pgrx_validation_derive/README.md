# PGRX Validation Derive

Procedural macros to dispatch the function output data type according to the features enabled.
