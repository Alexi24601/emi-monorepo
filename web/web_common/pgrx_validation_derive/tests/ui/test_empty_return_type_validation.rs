//! Submodule for testing compile failure of the `validation` attribute on functions which
//! do not return nothing.
use pgrx_validation_derive::validation;

#[validation]
/// A simple validation function which does not return anything should fail.
pub fn is_not_empty(_arg: &str) {}

/// The main for this test.
fn main() {}
