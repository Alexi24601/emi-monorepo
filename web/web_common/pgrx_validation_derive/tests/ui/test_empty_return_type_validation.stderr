error: Function `is_not_empty` must return a `Result<(), validation_errors::SingleFieldError>` or `Result<(), validation_errors::DoubleFieldError>` to be decorated with `validation`, but returns `()`.
 --> tests/ui/test_empty_return_type_validation.rs:5:1
  |
5 | #[validation]
  | ^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `validation` (in Nightly builds, run with -Z macro-backtrace for more info)
